src/diagnostics/build.rs

```diff
@@ -22,21 +22,59 @@ pub fn collect_build_diagnostics(
         diagnostics
     });
 
-    let root_document = workspace.documents_by_uri.values().find(|document| {
-        document.data.as_latex().map_or(false, |data| {
-            !document.uri.as_str().ends_with(".aux")
-                && data
-                    .extras
-                    .implicit_links
-                    .log
-                    .iter()
-                    .any(|u| u.as_ref() == build_log_uri)
-        })
-    })?;
+    let root_document_uri = Arc::clone(
+        &workspace
+            .documents_by_uri
+            .values()
+            .find(|document| {
+                document.data.as_latex().map_or(false, |data| {
+                    !document.uri.as_str().ends_with(".aux")
+                        && data
+                            .extras
+                            .implicit_links
+                            .log
+                            .iter()
+                            .any(|u| u.as_ref() == build_log_uri)
+                })
+            })?
+            .uri,
+    );
+
+    let build_log_uri = Arc::clone(&build_log_document.uri);
 
-    let base_path = PathBuf::from(root_document.uri.path());
+    let mut base_path = PathBuf::from(root_document_uri.path());
```

Comment:
You can use `root_document_uri.to_file_path()` here. The other method could cause problems on Windows.